### PR TITLE
ARROW-1371: [Website] Add "Powered By" page to the website

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: site/powered_by.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+## Powered By
+
+### Project and Product names using "Apache Arrow"
+
+Organizations creating products and projects for use with Apache Arrow, along
```
Comment (xhochy) [inline site/powered_by.md]: I used this paragraph from Spark but extended the wording to `Apache Arrow`. Spark was also talking about `Spark` alone but from my understanding they have a "more explicit trademark" on that then Arrow has.

**Comment by xhochy** [comment]
@pcmoritz @robertnishihara Can you give me a short line for Ray to include here?

**Comment by xhochy** [comment]
@elahrvivaz Can you give a line for Geomesa to include here?

**Comment by xhochy** [comment]
@jacques-n Can you give me a line for Dremio and one for dremio-oss?

**Comment by xhochy** [comment]
@BryanCutler @icexelloss Can you give me a line for Spark?

**Comment by xhochy** [comment]
@wesm @cpcloud Can you also give me a line for TwoSigma?

**Comment by xhochy** [comment]
Sorry for pining you all, I expect that the descriptions the individual authors write are much better than what I would come up with. I want to merge this before the 0.8.0 release and might add some preliminary ones if not all ready (but that should still be some days/weeks away).

**Comment by pcmoritz** [comment]
* **[Ray][5]:** A flexible, high-performance distributed execution framework with a focus on machine learning and AI applications. Uses Arrow to efficiently store Python data structures containing large arrays of numerical data in shared memory.

[5]: https://github.com/ray-project/ray

**Comment by wesm** [comment]
@pcmoritz want to mention also that a plasma originated from Ray?

**Comment by icexelloss** [comment]
I will give this a try:
```
* **[Apache Spark] [6]: ** Apache Spark™ is a fast and general engine for large-scale data processing. Spark (PySpark) uses Arrow to efficiently transfer data between Java and Python. 

[6]: https://spark.apache.org/ 
```

**Comment by wesm** [comment]
@icexelloss we should mention specifically the UDF use case and `toPandas`. Isn't it also true that Spark uses Arrow format for its internal in-memory columnar? per SPARK-20960 cc @cloud-fan

**Comment by icexelloss** [comment]
Good point. How about this?

**[Apache Spark] [6]: ** Apache Spark™ is a fast and general engine for large-scale data processing. Spark uses Apache to (1) improve performance of conversion between Spark DataFrame and pandas DataFrame (2) enable a set of vectorized user-defined functions (pandas_udf) in PySpark
[6]: https://spark.apache.org/ 

I will let @cloud-fan comment on if we should include `ColumnarVector` API here.

**Comment by cloud-fan** [comment]
> Spark uses Arrow format for its internal in-memory columnar

No, Spark only uses Arrow format when talking to Python, and may use Arrow to talk to more external systems in the future. The `ColumnVector` is still an internal interface, let's not include it. I think what @icexelloss wrote is good.

**Comment by jacques-n** [comment]
How about this for Dremio?

* **[Dremio][]:** A self-service data platform. Dremio makes it easy for users to discover, curate, accelerate, and share data from any source. It includes a distributed SQL execution engine based on Apache Arrow. Dremio reads data from any source (RDBMS, HDFS, S3, NoSQL) into Arrow buffers, and provides fast SQL access via ODBC, JDBC, and REST for BI, Python, R, and more (all backed by Apache Arrow).

**Comment by elahrvivaz** [comment]
thanks, here's a blurb for geomesa:
```
[GeoMesa](https://github.com/locationtech/geomesa) is a suite of tools that enables large-scale geospatial query and analytics on distributed computing systems. GeoMesa supports query results in the Arrow IPC format, which can then be used for in-browser visualizations and/or further analytics.
```

**Comment by BryanCutler** [comment]
+1 on what Li wrote for Spark, thanks!

**Comment by wesm** [comment]
I added a few more projects: MapD (cc @tmostak), libgdf (cc @seibert), QuiltData (cc @kevinemoore), pandas, Dask (cc @mrocklin). Happy to edit these descriptions as desired.

Any more projects out there we are aware of? Maybe we should add https://github.com/red-data-tools, @kou?

File: site/powered_by.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+[6]: https://ray-project.github.io/2017/08/08/plasma-in-memory-object-store.html
+[7]: https://spark.apache.org/
+[8]: https://github.com/locationtech/geomesa
+[9]: https://www.dremio.com/
+[10]: https://github.com/mapd/mapd-core
+[11]: https://gpuopenanalytics.com/
+[12]: https://pandas.pydata.org
+[13]: https://quiltdata.com/
+[14]: https://github.com/gpuopenanalytics/libgdf
+[15]: https://github.com/dask/dask
```
Comment (mrocklin) [inline site/powered_by.md]: For Dask I recommend pointing to the docs at https://dask.pydata.org

**Comment by kou** [comment]
Yes please. How about the following entry?

```markdown
* **[Red Data Tools][16]:** A project that provides data processing
  tools for Ruby. It provides [Red Arrow][17] that is a Ruby bindings
  of Apache Arrow based on Apache Arrow GLib. Red Arrow is a core
  library for it. It also provides many Ruby libraries to integrate
  existing Ruby libraries with Apache Arrow. They use Red Arrow.

[16]: https://red-data-tools.github.io/
[17]: https://github.com/red-data-tools/red-arrow/
```

**Review by wesm — APPROVED** [review]
+1. We can accept more additions in follow up PRs. Thanks all!

**Pull request closed without merging**